app/assets/javascripts/discourse/app/components/modal/bulk-topic-actions.gjs

```diff
@@ -151,16 +151,24 @@ export default class BulkTopicActions extends Component {
         break;
       case "archive_messages":
       case "move_messages_to_inbox":
+        let params = { type: this.model.action };
+
         let userPrivateMessages = getOwner(this).lookup(
           "controller:user-private-messages"
         );
 
-        let params = { type: this.model.action };
-
         if (userPrivateMessages.isGroup) {
           params.group = userPrivateMessages.groupFilter;
         }
 
+        let groupPrivateMessages = getOwner(this).lookup(
+          "controller:group-messages"
+        );
+
+        if (groupPrivateMessages.isGroup) {
+          params.group = groupPrivateMessages.model.name;
+        }
```

Comment:
This is a pretty indirect way of doing it, but it matches exactly the approach used for the user page. Since I'm focused on fixing the immediate issue, I consider it out of scope to change the approach in the same PR, and instead I'm aiming for consistency with the existing system. 🙏